### PR TITLE
add missing return types and enforece via linter

### PR DESCRIPTION
**Pull request by foxriver76**

### CODE REVIEW

File: packages/common/src/lib/common/maybeCallback.test.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { maybeCallback, maybeCallbackWithError, maybeCallbackWithRedisError } fr
 
 async () => {
     // maybeCallbackWithError, empty callback
-    const cb = () => {};
+    const cb = (): void => {};
```
[the remark this answers is not in the document]
Comment (foxriver76) [reply packages/common/src/lib/common/maybeCallback.test.ts:11]: I explicitly added it for tests files I guess, from my pov it also does not hurt in tests. Maybe you have made opposite experience?

File: packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts [hunk 2]
```diff
@@ -4291,7 +4296,7 @@ export class ObjectsInRedisClient {
         params?: ioBroker.GetObjectViewParams,
         options?: CallOptions,
         callback?: ioBroker.GetObjectViewCallback<any>
-    ) {
+    ): Promise<ioBroker.CallbackReturnTypeOf<ioBroker.GetObjectViewCallback<any>> | void> {
```
[the remark this answers is not in the document]
Comment (foxriver76) [reply packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts:4316]: this and the above ones are a bit special on the db context, as they can return promise as well as cb and I haven't found a nice solution to pass like `ioBroker.GetObjectViewPromise | Promise<void>` to it, as he thens ays async return type needs to be the global promise, thus everything needs to be wrapped in a `Promise<typeStuff>`. You have a solution to this, w/o creating new types for the mixed (cb or the "real" promise) return.

File: packages/common/src/lib/common/tools.ts [hunk 3]
```diff
@@ -2198,7 +2196,7 @@ export function getConfigFileName(): string {
  * @param argsObj An `arguments` object as passed to a function
  * @param startIndex The optional index to start taking the arguments from
  */
-function sliceArgs(argsObj: IArguments, startIndex = 0) {
+function sliceArgs(argsObj: IArguments, startIndex = 0): any[] {
```
[the remark this answers is not in the document]
Comment (foxriver76) [reply packages/common/src/lib/common/tools.ts:2199]: It is used by the `promisify` function, which basically uses all any types, so shouldn't hurt to be honest. Also technically, rest args are passed to a function, and then processed by this method, so not sure if we can infer anything useful.

File: packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts:4316]: got it

File: packages/common/src/lib/common/maybeCallback.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlCalzone) [reply packages/common/src/lib/common/maybeCallback.test.ts:11]: I don't really care, just found that having to annotate mocks etc. in tests is a bit annoying.

**Pull request merged**